Source/WebKit/GPUProcess/graphics/WebGPU/RemoteDevice.cpp

```diff
@@ -151,9 +156,21 @@ void RemoteDevice::importExternalTexture(const WebGPU::ExternalTextureDescriptor
 {
     auto convertedDescriptor = m_objectHeap.convertFromBacking(descriptor);
     ASSERT(convertedDescriptor);
-    if (!convertedDescriptor)
+    if (!convertedDescriptor || !convertedDescriptor->mediaIdentifier)
         return;
 
+    m_remoteGPU->remoteRenderingBackend().performWithMediaPlayerOnMainThread(WebCore::MediaPlayerIdentifier(convertedDescriptor->mediaIdentifier), [&convertedDescriptor](auto& player) mutable {
```

Comment:
Is this synchronous? We need to make sure it's synchronous

Reply:
It is :)

```
void RemoteRenderingBackend::performWithMediaPlayerOnMainThread(MediaPlayerIdentifier identifier, Function<void(MediaPlayer&)>&& callback)
{
    callOnMainRunLoopAndWait
```